File: Companies/Google/MissingNumber.java
package Companies.Google;

public class MissingNumber {
    public int missingNumber(int[] nums) {
        int re = nums.length;
        for (int i = 0; i < nums.length; i++) {
            re ^= i ^ nums[i];
        }
        return re;
    }
}

File: Companies/VMware/MissingNumber.cpp
#include <iostream>
#include <vector>

using namespace std;

class MissingNumber
{
public:
  int missingNumber(vector<int> &nums)
  {
    int re = nums.size();
    for (int i = 0; i < nums.size(); i++)
    {
      re ^= i ^ nums[i];
    }
    return re;
  }

  int missingNumberII(vector<int> &nums)
  {
    int sum = (nums.size()) * (nums.size() + 1) / 2;
    for (auto num : nums)
    {
      sum -= num;
    }
    return sum;
  }
};